### PR TITLE
Update sklearn example

### PR DESCRIPTION
**Pull request by danielfrg**
Updating the scikit-learn sample app:

- Updated the workflow to the new UI, forms and stuff
- Added instructions on the environment selection
- Updated the dependencies so it works with the python 3.7 environment
- Changed the serialisation format to joblib, this is the recommended by scikit-learn
- Added two query options, real time predict and batch based on a file

### CODE REVIEW

**Comment by koverholt** [comment]
Thanks, LGTM!

In the current/older version with the pickle file, I ended up pinning `numpy` and `scipy` in the `requirements.txt` for maximum reproducibility. Do you see a need to do that here as well?

Also, thanks for including the specific language and environment to use!

File: algo-dev-demo/scikit-learn-demo/README.md [hunk 1]
```diff
@@ -1,67 +1,93 @@
...
-- Note that you'll need to change the name of the data collection to the one we created earlier. 
+- Open the file with the same name as the algorithm, on this case it would be: `scikit_learn_demo.py`
+- Delete that code, and copy and paste the code from the file [src/boston-housing-prices.py](https://github.com/algorithmiaio/sample-apps/blob/master/algo-dev-demo/scikit-learn-demo/src/boston-housing-prices.py)
+- Change the name of the data collection to the one we created earlier.
 
-Recall our data collection is called "scikit_learn_demo" and you'll need
-to change "YOUR_USERNAME" to your own username: `file_path =
-'data://YOUR_USERNAME/scikit_learn_demo/scikit-demo-boston-regression.pkl'`
+Recall our data collection is called `scikit_learn_demo` and you'll also need
+to change `USERNAME` to your own username on this line:
```
Comment (koverholt) [inline algo-dev-demo/scikit-learn-demo/README.md:41]: Suggest changing "username" to "username or organization".

**Comment by danielfrg** [comment]
Yeah, I think that makes sense, I will pin `numpy`. Is there a reason you included `scipy`?

**Comment by koverholt** [comment]
I include `numpy` and `scipy` both since they are deps of `sklearn`. In theory pinning only `sklearn` would handle this, but you know how finicky these things can be.

File: algo-dev-demo/scikit-learn-demo/README.md [hunk 2]
```diff
@@ -1,67 +1,94 @@
...
+
+```
+file_path = "data://USERNAME/scikit_learn_demo/scikit-demo-boston-regression.pkl"
+```
+
+Be sure to save this changes.
```
Comment (koverholt) [inline algo-dev-demo/scikit-learn-demo/README.md:47]: "this" -> "these"

File: algo-dev-demo/scikit-learn-demo/README.md (hunk 1 above)
Comment (danielfrg) [reply algo-dev-demo/scikit-learn-demo/README.md]: done

File: algo-dev-demo/scikit-learn-demo/README.md (hunk 2 above)
Comment (danielfrg) [reply algo-dev-demo/scikit-learn-demo/README.md]: ty!

**Comment by danielfrg** [comment]
Thanks for the review! added scipy pin and fixes your suggestions

**Review by koverholt — APPROVED** [review]

**Review by bnlb — APPROVED** [review]
lgtm!

**Pull request merged**